source/screens/caseScreens/CaseSummary.js

```diff
@@ -22,35 +23,87 @@ const SummaryHeading = styled(Text)`
   margin-bottom: 16px;
 `;
 
+const computeCaseCardComponent = (caseData, form, colorSchema, navigation) => {
+  const {
+    status,
+    currentPosition: { currentMainStep: currentStep } = {},
+    details: { period: { endDate } = {} } = {},
+  } = caseData;
+  const totalSteps = form?.stepStructure?.length || 0;
+  const applicationPeriodMonth = getSwedishMonthNameByTimeStamp(endDate, true);
+
+  if (status?.type?.includes('ongoing')) {
+    return (
+      <Card colorSchema={colorSchema}>
+        <Card.Body shadow color="neutral">
+          <Card.Title colorSchema="neutral">{applicationPeriodMonth}</Card.Title>
+          <Card.SubTitle>
+            Steg {currentStep} / {totalSteps}
+          </Card.SubTitle>
+          <Card.Progressbar currentStep={currentStep} totalStepNumber={totalSteps} />
+          <Card.Text>{status.description} </Card.Text>
+          <Card.Button
+            onClick={() => {
+              navigation.navigate('Form', { caseId: caseData.id });
+            }}
+          >
+            <Text>Fortsätt ansökan</Text>
+            <Icon name="arrow-forward" />
+          </Card.Button>
+        </Card.Body>
+      </Card>
+    );
+  }
+
+  if (status?.type?.includes('notStarted')) {
+    return (
+      <Card colorSchema={colorSchema}>
+        <Card.Body shadow color="neutral">
+          <Card.Title colorSchema="neutral">{applicationPeriodMonth}</Card.Title>
+          <Card.SubTitle>{status.name}</Card.SubTitle>
+          <Card.Text>{status.description} </Card.Text>
+          <Card.Button
+            onClick={() => {
+              navigation.navigate('Form', { caseId: caseData.id });
+            }}
+          >
+            <Text>Starta ansökan</Text>
+            <Icon name="arrow-forward" />
+          </Card.Button>
+        </Card.Body>
+      </Card>
+    );
+  }
+
+  return (
+    <Card colorSchema={colorSchema}>
+      <Card.Body shadow color="neutral">
+        <Card.Title colorSchema="neutral">{applicationPeriodMonth}</Card.Title>
+        <Card.SubTitle>{status.name}</Card.SubTitle>
+        <Card.Text>{status.description} </Card.Text>
+      </Card.Body>
+    </Card>
+  );
```

Comment:
Same thing here, this feels perhaps a little redundant; essentially all the code in the last return statement is repeated 3 times. But it might be a bit more readable, on the other hand.

Reply:
Agree, the Card, Card.Body, Card.Tiitle, Card.Subtitle and Card.Title only needs to be defined one time. Perhaps
```
 const buttonProps = {}
 if (...) {
   // pupulate buttonProps
 }
if (...) {
 // populate buttonProps.
}
 return (
 <Card colorSchema={colorSchema}>
      <Card.Body shadow color="neutral">
        <Card.Title colorSchema="neutral">{applicationPeriodMonth}</Card.Title>
        <Card.SubTitle>{status.name}</Card.SubTitle>
        <Card.Text>{status.description} </Card.Text>
        { buttonProps && <Card.Button
            onClick={buttonProps.onClick}
          >
            <Text>{buttonProps.text}</Text>
            <Icon name={buttonProps.iconName} />
          </Card.Button>}
      </Card.Body>
    </Card>)
   }
```

Reply:
Did not take into account the SubTitle, but you get the idea

Reply:
Sorry my brain is a bit tierd, this is not relevant since you already have addressed these fixes. My bad :)